src/Command/Argument/Search/CreateArguments.php

```diff
@@ -0,0 +1,656 @@
+
+    /**
+     * Filters the results to those appearing only in specific attributes of the document, like title or URL.
+     *
+     * @param  array $fields
+     * @return $this
+     */
+    public function inFields(array $fields): self
+    {
+        $this->arguments[] = 'INFIELDS';
+        $this->arguments[] = count($fields);
+        $this->arguments = array_merge($this->arguments, $fields);
+
+        return $this;
+    }
+
+    /**
+     * Limits the attributes returned from the document.
+     * Num is the number of attributes following the keyword.
+     * If num is 0, it acts like NOCONTENT.
+     * Identifier is either an attribute name (for hashes and JSON) or a JSON Path expression (for JSON).
+     * Property is an optional name used in the result. If not provided, the identifier is used in the result.
+     *
+     * If you want to add alias property to your identifier just add "true" value in identifier enumeration,
+     * next value will be considered as alias to previous one.
+     *
+     * Example: 'identifier', true, 'property' => 'identifier' AS 'property'
+     *
+     * @param  int         $count
+     * @param  string|bool ...$identifier
+     * @return $this
+     */
+    public function addReturn(int $count, ...$identifier): self
+    {
+        $arguments = func_get_args();
+
+        $this->arguments[] = 'RETURN';
+
+        for ($i = 1, $iMax = count($arguments); $i < $iMax; $i++) {
+            if (true === $arguments[$i]) {
+                $arguments[$i] = 'AS';
+            }
+        }
+
+        $this->arguments = array_merge($this->arguments, $arguments);
+
+        return $this;
+    }
+
+    /**
+     * Returns only the sections of the attribute that contain the matched text.
+     *
+     * @param  array  $fields
+     * @param  int    $frags
+     * @param  int    $len
+     * @param  string $separator
+     * @return $this
+     */
+    public function summarize(array $fields = [], int $frags = 0, int $len = 0, string $separator = ''): self
+    {
+        $this->arguments[] = 'SUMMARIZE';
+
+        if (!empty($fields)) {
+            $this->arguments[] = 'FIELDS';
+            $this->arguments[] = count($fields);
+            $this->arguments = array_merge($this->arguments, $fields);
+        }
+
+        if ($frags !== 0) {
+            $this->arguments[] = 'FRAGS';
+            $this->arguments[] = $frags;
+        }
+
+        if ($len !== 0) {
+            $this->arguments[] = 'LEN';
+            $this->arguments[] = $len;
+        }
+
+        if ($separator !== '') {
+            $this->arguments[] = 'SEPARATOR';
+            $this->arguments[] = $separator;
+        }
+
+        return $this;
+    }
+
+    /**
+     * Formats occurrences of matched text.
+     *
+     * @param  array  $fields
+     * @param  string $openTag
+     * @param  string $closeTag
+     * @return $this
+     */
+    public function highlight(array $fields = [], string $openTag = '', string $closeTag = ''): self
```

Comment:
Are you intentionally not separating the arguments of FT.CREATE and FT.SEARCH into two separate classes?

Reply:
But according to what I see in the form it is now, you can insert highlight into the FT.CERATE command even though it does not have this parameter. Am I missing something?

Reply:
This PR is about FT.SEARCH, but it's based on FT.CREATE branch, that's why there's both changes here. This parameter for FT.SEARCH.